data_safe_haven/infrastructure/programs/sre/workspaces.py

```diff
@@ -193,13 +193,17 @@ def __init__(
         file_uploads = [
             (FileReader(resources_path / "workspace" / "run_all_tests.bats"), "0444")
         ]
+        vm_index = 0
         for test_file in pathlib.Path(resources_path / "workspace").glob("test*"):
             file_uploads.append((FileReader(test_file), "0444"))
         for vm, vm_output in zip(vms, vm_outputs, strict=True):
```

Comment:
```suggestion
        for vm_index, (vm, vm_output) in enumerate(zip(vms, vm_outputs, strict=True)):
```
you can do the indexing as you loop in Python by using `enumerate`